pkg/services/librarypanels/api.go

```diff
@@ -15,18 +15,17 @@ func (lps *LibraryPanelService) registerAPIEndpoints() {
 	}
 
 	lps.RouteRegister.Group("/api/library-panels", func(libraryPanels routing.RouteRegister) {
-		libraryPanels.Post("/", middleware.ReqSignedIn, binding.Bind(addLibraryPanelCommand{}), api.Wrap(lps.addLibraryPanelEndpoint))
+		libraryPanels.Post("/", middleware.ReqSignedIn, binding.Bind(AddLibraryPanelCommand{}), api.Wrap(lps.createHandler))
 	})
 }
 
-// addLibraryPanelEndpoint handles POST /api/library-panels.
-func (lps *LibraryPanelService) addLibraryPanelEndpoint(c *models.ReqContext, cmd addLibraryPanelCommand) api.Response {
-	cmd.OrgId = c.SignedInUser.OrgId
-	cmd.SignedInUser = c.SignedInUser
+// createHandler handles POST /api/library-panels.
+func (lps *LibraryPanelService) createHandler(c *models.ReqContext, cmd AddLibraryPanelCommand) api.Response {
+	panel, err := lps.Create(c, cmd)
```

Comment:
I don't see the point of this change. Why add another indirection with more code? Is there any upside? The downside is that it's more difficult to review and verify the code. The benefits of an abstraction should outweigh its cost (every abstraction has a cost).

I think the `Result` field should be removed from the command though. IMO the command should only contain parameters from the HTTP client, and a normal return value should be used to pass the result back.

Reply:
I totally agree about the command, that was something I copied from `ngalert` as I thought that was a good example, I will change that too. As for the point of the change, I wrote in the other comment.

Reply:
https://github.com/grafana/grafana/pull/29642#discussion_r538283212

Reply:
@aknuds1 we are going to have a provisioning workflow that will also create panels, so we need a function that create a library panels that is shared between api and internal usage

Reply:
Torkel I've had a chat with Arve and I'll add it when we need it instead. (YAGNI)

Reply:
Yeah, it's no bother to add a method when it's needed, it's more difficult to envision the right design beforehand.